### ISSUE TITLE
[Android] deviceready has not fired after 5 seconds. 

### ISSUE THREAD

**Issue by fidoboy** [AppGyver/steroids#772]
It happens with some webViews, not sure why... chrome console displays this:
```
    deviceready has not fired after 5 seconds. cordova.js:1154
    Channel not fired: onCordovaConnectionReady cordova.js:1147
    Channel not fired: onCordovaInfoReady cordova.js:1147
    Channel not fired: onFileSystemPathsReady cordova.js:1147
```

**Comment by aromanyuk** [comment]
same thing

**Comment by doomboy00** [comment]
+1

**Comment by pre** [comment]
Can you find any pattern or steps how this could be reproduced?

**Comment by fidoboy** [comment]
Nope. Not sure about when it happens, I haven't found any pattern yet; it works on some webViews and it displays the warnings on others

**Comment by fidoboy** [comment]
Just created a new build using the latest 4.0.5-edge2 released today and now i get:

```
deviceready has not fired after 5 seconds. cordova.js:1154
Channel not fired: onFileSystemPathsReady 
```

on my initialView and app does not start, it hangs on splash screen... It was working before this release :disappointed:

**Comment by fidoboy** [comment]
Reloading the page using Chrome remove inspector solves the problem; the page loads fine and deviceready message doesn't appear

**Comment by pre** [comment]
So this particular message about `onFileSystemPathsReady` started appearing only after 4.0.5-edge2 (which updates the Cordova File plugin, which is the probable cause for it).

**Comment by fidoboy** [comment]
See my first message, it happens also on previous version. Weird thing is that I'm not using that plugin on this particular view

**Comment by fidoboy** [comment]
I've right tested it again using new build with 4.0.5-edge3 and still same results. On my initial view (and many others also) i get:

```
deviceready has not fired after 5 seconds. cordova.js:1154
Channel not fired: onFileSystemPathsReady 
```

and again, reloading the page from chrome remote inspector, solves the problem and the page loads and works fine.

**Comment by fidoboy** [comment]
As additional info i should add that i'm not using the file transfer plugin or any other related to files on this view. My procedure is this:

```
angular.element(document).ready(function() {
    function onDeviceReady() {
        document.removeEventListener("deviceready", onDeviceReady, false);
        var networkState = navigator.connection.type;
        ......
    }
    document.addEventListener("deviceready", onDeviceReady,false);
}
```

**Comment by pre** [comment]
@fidoboy I meant to ask: What is the last Scanner version that was working for you?

**Comment by pre** [comment]
@fidoboy Ah, didn't read your code carefully. 

I'm pretty sure the problem you're facing comes from the fact that you are trying to catch `deviceready` inside an Angular function. This is not a reliable way to do it.

This is what 99% certainly will happen:

- your application starts to execute both Angular and Cordova Javascript files simultaneously
- Angular will take it's time loading, after a while it'll end up binding your `onDeviceReadyFunction`
- Cordova will load *at the same time*.
--> If Cordova loads faster than your Angular code, then the `deviceready` event will be fired *before*  you have placed your event listener to catch it. 

Ie. In your current code it's not guaranteed that your event listener is not alive when `deviceready` is fired.

--> You must bind the `deviceready` listener *before* you start loading `http://localhost/cordova.js`. If you use [`supersonic.device.ready`](http://docs.appgyver.com/supersonic/api-reference/stable/supersonic/device/ready/) this will be handled for you (just load supersonic.js before cordova.js).

Unless you can provide a code reproduction that does not work like your code above, I will deem this as a bug in your application.

**Comment by sebmartel** [comment]
+1

**Comment by mmerkes** [comment]
@pre I don't believe that the issue is that Cordova loads faster than Angular for the following reasons:

1. `deviceready has not fired after 5 seconds.` is being called from Cordova, not Angular
2. If I insert a `document.addEventListener("deviceready", function () { console.log('DEVICE READY FIRE'); }, false);` directly into `index.html` immediately before or after we include the cordova file, it never gets fired.

**Comment by fidoboy** [comment]
A key point in the answer is this line from the documentation of the deviceready event.

```
This event behaves differently from others in that any event handler registered after the event has been fired will have its callback function called immediately.
What this means is that you won't 'miss' the event if you add a listener after the event has been fired.
```

So in theory my approach is valid and it works fine on iOS. May be that it's a cordova bug in latest android releases however... What i need is to make some operations with DOM so i need both conditions: that DOM is ready and also cordova deviceready. Can't see any other way to solve it then... may be using timers and loops, but it looks pretty ugly to me. 

Anyway, if others are having the same problem and they are not using angular or my approach using the document DOM ready... may be that there is some kind of bug on steroids.

**Comment by mmerkes** [comment]
Also, for us, iOS works fine. The problem is Android only. And refreshing via the Chrome console does not fix the issue for us.

**Comment by fidoboy** [comment]
I believe that it happens when the page loads too fast... on Android, some webViews are working fine on my app, and this problem does not happen, but it fails always on others.

**Comment by fidoboy** [comment]
What i know is that the `deviceready has not fired after 5 seconds` was not happening on previous versions on my initialView which was using exactly the same code that i've described before, without problems. The message was causing problems on other webViews, but not on this one until 4.0.5-edge2.  So now, after this release, the problem seems to be more frequent and noticeable.

**Comment by fidoboy** [comment]
@mmerkes: sometimes I must reload several times the page until it works; not sure if it's also your case.

**Comment by mmerkes** [comment]
Did further research, and the problem is definitely related to a change between 4.0.5-edge1 and 4.0.5-edge3. If I build using 4.0.5-edge1, it works just fine.

**Comment by fidoboy** [comment]
From my experience, it was happening before this version also, but it was not so noticeable. In my app I was having this problem in a few webViews also. But as you pointed, with these latest versions it happens almost all the time and in almost all webViews

**Comment by mmerkes** [comment]
I've narrowed down the problem further. If I force the 4.0.5-edge3 build to use the cordova.js from 4.0.5-edge1, everything works fine, so the issue is related to some change in cordova.js between the two builds, but I have dug down to see what the offensive piece of code is yet.

**Comment by fidoboy** [comment]
👏 great @mmerkes!! I suspected that as I pointed in my previous comment. The issue is that for some unknown reason, the event trigger is buggy and doesn't work as expected or may be that steroids native part is interfering in some way...

Let's hope that your discovering could help to appgyver team to locate and solve the bug

**Comment by mmerkes** [comment]
@fidoboy Yeah, something is killing the event trigger. @pre Suggested the cordova change as well, and hopefully they'll be able to resolve this quickly because we're trying to get into the app store ASAP, and this is holding us up. I guess we'll have to leave it to them for now...

**Comment by pre** [comment]
@mmerkes this should not prevent you from releasing if the current stable 4.0.4 does not have this issue. (Thanks for the trust btw in case you are ready to release your app with an edge version of the runtime :)

I'll look into this on Monday.

**Comment by pre** [comment]
1) Issue seems to be that the `deviceready` event has slowed down when using the Initial View. On Nexus 4 it taks consistently between 3,2 - 4,2 seconds to be fired on each Tab after dismissing the Initial View. If no initial view is used, then the `deviceready` event will be fired under a second.

On a slower device this will be even slower, which will launch the timed message about `deviceready has not fired after 5 seconds`. Eg Samsung Galaxy Tag 3 (Android 4.2) takes 5,3 - 5,5 seconds to get `deviceready`. The event, however, eventually comes and there's no disfunction.

This slowness is consistent between versions 4.0.3, 4.0.4, and 4.0.5-edge3. Platform webview appears to load faster than the Crosswalk runtime.


2) File `cordova.js` has changed between 4.0.5-edge versions because by default all plugin javascripts are merged into `cordova_merged.js`. Internally this file is served in stead of `cordova.js`. (It will significantly speed up loading since otherwise there'd be dozens of  load requests, one for each plugin).


3) I haven't been able to reproduce the problem where `deviceready` event does not get fired at all. Although you get the timed out warning of not launching after 5 seconds, does your application disfunction? Do you eventually get the `deviceready` event?

**Comment by pre** [comment]
I also tested with a 3-month-old Scanner 4.0.1 build (ie didn't rebuild, used to old APK). Deviceready takes ~5 seconds after Initial View on Samsung Galaxy Tab 3 (Android 4.2) using Crosswalk.

**Comment by mmerkes** [comment]
Thanks for your research @pre . I'm doing most of my testing on a Galaxy S4. I'll go over some of the steps I went through to investigate.

At first, I thought that it might just be taking longer than 5 seconds, so I wanted to add a longer timeout. First, I added a custom listener to see if it would fire, and it never happened. Then, when I forced it to use a copy of cordova.js and adjusted the timeout up to 20 seconds, I would get the same log message. Anything plugin-related in our application is broke.

I did some initial attempts to get a bare-bones repo, and stripped everything out of our application. However, if I stripped everything, I don't get the same issue (i.e. everything but cordova, our libraries, and main controller), which makes me believe it's likely one of two scenarios:

1. The timeout gets exponentially slower on a larger applications.
2. Something inside of our application is interfering with the new cordova file that didn't bother the old one.

Did the API change at all with cordova? The version numbers listed in the files are the exact same, so it'd be helpful if that would have been bumped, given the notable changes.

**Comment by mmerkes** [comment]
Also, when we remove all of the plugins, it still doesn't fire.

**Comment by fidoboy** [comment]
I'm using two devices mainly: Xiaomi MiPad and Samsung Galaxy Note II. Both having the same problem. deviceready event should fire even if it's called 20 seconds after cordova is loaded, but it's not.

**Comment by poppahorse** [comment]
+1

Seem to have had this problem on and off since I started using supersonic. Have definitely had it on scanner builds prior to 4.0.5-edge1.

Unsure if related but seem to be getting other sporadic problems
- sporadically get `steroids is not defined`

- sporadically get `processMessage failed: Stack: TypeError: undefined is not a function` on `pushNotification.onMessageInBackground(` for push plugin

- sporadically get `facebookConnectPlugin is not defined` on my initialview

which all seem to involve stuff not getting loaded in time / the right order?

**Comment by fidoboy** [comment]
From my own experience, it seems that you have a mix of several different things. I can confirm the problem with `deviceready not being fired` on Android platform, but i've no issues with PushNotifications plugin (using this https://github.com/AppGyver/PushNotifications) or Facebook add-on (it doesn't work on Android when Facebook app is installed however). `Steroids not defined` seems to point to a incorrect loading order or other timed related issues in your code.

**Comment by poppahorse** [comment]
The PushNotifications plugin works, however because it needs binding to `deviceready` sometimes its not returning a token, and it doesn't seem to want to recognise the background or foreground events, or the fact that the plugin is defined, presumably because or the `deviceready` issues.

I am going to try to catch `deviceready` outside of angular and see if that makes any difference.

**Comment by fidoboy** [comment]
I've tried it already and it fails the same. I'm almost sure that it's not angular related issue. The deviceready event should fire anywhere you call it, even if cordova.js has being loaded before... this event is a special case and it doesn't work like other events. Anyway, try it and share your experience...

**Comment by pre** [comment]
* The `steroids is not defined` is a combinative issue with Supersonic and post messages. Supersonic sends lots of events in the background using post messages. Sometimes it happens that `steroids.js` is not (yet) loaded when the post message already arrives and there's a reference to the `steroids` variable. I have a reproduction case for that. 

* For PushNotification specific plugin issues, please create a ticket in https://github.com/AppGyver/PushNotifications/issues

* The investigation by @mmerkes gives a clue, that this might have something to do with all the Cordova javascript being merged into one single file. This was introduced in [4.0.3](https://github.com/AppGyver/scanner/blob/master/changelog-android-fresh.md) to speed up the `deviceready`.. 

@mmerkes and @fidoboy  If you try your app with Android 4.0.2 Scanner, do you have an issue with the `deviceready` ? If `deviceready` fires on 4.0.2 for you, that would pinpoint this problem to the change of merging all the Cordova javascripts into one file.

**Comment by fidoboy** [comment]
From my POV this event works like a filter, to ensure that device is ready and cordova is loaded before calling your plugins or other cordova related code. If it's being called after cordova is loaded and device is ready, then it works like: ìf (cordova device is ready) {... your code here ...}` in other case it waits until cordova finishes loading and then executes your callback

**Comment by fidoboy** [comment]
@pre i've tried with 4.0.2 scanner and my initialView works fine, i've issues with other webViews however, but it seems to be less frequent than when using 4.0.5. I'll investigate a bit more on my code to see if 'deviceready not fired' messages on 4.0.2 are related to something wrong in my code or if it's the same problem...

**Comment by fidoboy** [comment]
NOTE: I'm testing on android 4.2 and sometimes also with 4.4 on other device

**Comment by fidoboy** [comment]
I think that i've made a discovering with this issue. Using:

```
<script src="/cordova.js"></script>
```

instead of:

```
<script src="http://localhost/cordova.js"></script>
```

seems to fix the issue, at least in my initialView. Now it seems that it loads always and no `deviceready not fired` messages anymore. Not sure if removing ´http://localhost` from the path will lead into another problems (I believe that i added that in the past due to some problems in Steroids with Cordova on Android platform, but can't remember right now what it where about). In my current code, cordova.js is the only script being loaded adding ´http://localhost`

**Comment by fidoboy** [comment]
Not sure if it's related, but right now i'm dealing with a very very weird issue. When i load a webView from drawer to replace center view it works fine, but when i load the same web view from current webView using push the webView loads empty (white page, DOM items loaded but not visible) and it only works when i reload it from chrome inspector. This same code is working fine in both cases on iOS... this is driving me nuts :laughing:

**Comment by fidoboy** [comment]
ok, very off topic here... but problem was `window.innerWidth` or `window.innerHeight` not working on Android on DOM ready :weary:

**Comment by jeffthompson1971** [comment]
So any update on what will happen with this issue?  i'm hitting it all the time.  nearly 100% of the time my main 'home' view loads this is what i see.  lots of good debugging on this thread just not seeing a conclusion that jumps out.. is there a best practice? order of script includes, wait for the next release?

**Comment by fidoboy** [comment]
AFAIK there is no solution yet. It seems that using old releases it does not happen (but you will suffer other limitations and/or bugs), so you can try this way. Other than that, there is nothing that you can do right now, just wait...

**Comment by jDeppen** [comment]
+1

**Comment by sebmartel** [comment]
@pre What's the current priority of this issue on the appgyver list?  We are waiting on this to move to the latest builds.

**Comment by poppahorse** [comment]
yeah this is really causing me issues still, any update on the issue would be appreciated. am also getting this 100% of the time, and for things such as push notifications where it relies on deviceready to get the device token, its just not firing

**Comment by bFlood** [comment]
+1

**Comment by pre** [comment]
From forums: https://forums.appgyver.com/#!/steroids:deviceready-has-not-fired-a

****

Yeah, this should be prioritized up. Sorry for letting you guys wait.

**Comment by fidoboy** [comment]
Well, it's a very serious issue in Android platform which is causing many problems directly or indirectly... Let us know if we can help with anything, providing more info or tests, etc

**Comment by fidoboy** [comment]
What I've found is that the issue happens randomly, it happens most of the times but sometimes it's not. The same webView, with same code, sometimes works at first time and many times it doesn't. In my tests, when I debug my code from scanner, modifying anything on my code and causing steroids to rebuild the code leads into a success, but if you don't change anything it fails frequently, so this leads me to think that it's related with the loading phase and some timers or wait conditions on the native side.

As i've mentioned in previous posts on this thread, `deviceready` event should work even if it is called after cordova is ready or inside DOM ready or document ready events. By reading the Cordova docs i've verified that this event is a special case and it should work this way.

**Comment by kaipk** [comment]
i have the same problem with a slow device. +1

**Comment by poppahorse** [comment]
As we know the problem is Android based, but to speed up development and debugging (taking 30+ seconds to load a view once you make a change is frustrating), I have just been accessing the views directly in a browser eg `http://192.168.0.50:4567/app/poll/index.html?id=2` and it loads instantly

**Comment by pre** [comment]
@poppahorse Are you using livereload in the client? That may cause some of these effects. For the investigation above, my assumption has been that livereload has not been used. Could you please confirm that? Thanks

**Comment by sebmartel** [comment]
bump bump!   Still a blocker for us to move to the latest edge builds.

**Comment by fidoboy** [comment]
It could be very useful also to have any doc or guide about all initialization events. I'm having problems with `deviceready`from Cordova, ```steroids.on('ready', function() {...});``` and document ready. I'm not sure about the right order to chain all these three events when you need to perform any operation on DOM tree but also you need to have Cordova and Steroids ready before it. Does steroids ready event works in the same way than Cordova `deviceready` event? I want to mean that if it's called from DOM ready and steroids is already loaded, will the callback work also?

**Comment by PeanutButterCo** [comment]
+1

**Comment by mertenhanisch** [comment]
+1

**Comment by neophob** [comment]
+1

**Comment by Chener** [comment]
+1 
It took me a whole day to get here and convinced myself to rollback to old phonegap version.

**Comment by harsaharsa** [comment]
I finally had time to investigate this, and while I could not reproduce the case that `deviceready` never fires, I was able to determine that the `cordova.js` load times are affected by the existence of other WebViews. Adding more preloaded WebViews affects the load time, and similarly removing `cordova.js` from other WebViews speeds up load times. This leads me to suspect that Cordova somehow ends up waiting for more than one WebView to finish loading everything before it fires the `deviceready` event.

Could someone share a Steroids project where the `deviceready` event consistently doesn't fire so I could investigate further?

Also, if you add a `document.reload()` function to a `DOMContentLoaded` event listener, does that get rid of the errors (we used to have a similar situation back with Legacy Android)?

**Comment by sebmartel** [comment]
Harry,

I'll share our project which a single view.

**Comment by harsaharsa** [comment]
Thanks to @sebmartel's repro project, I managed to track down the cause, which was a nasty race condition caused by a Crosswalk peculiarity. The issue only affected projects that loaded numerous separate assets (i.e. CSS/JS/image files etc), which is why it could not be replicated with a more simple project.

Android [4.0.5-edge7](https://github.com/AppGyver/scanner/blob/master/changelog-android-fresh-edge.md#405-edge7-2015-03-09) is now available from the Build Service with a fix for this.

Please verify that it works for you!

**Comment by fidoboy** [comment]
Great news @harsaharsa!! I'll try it ASAP. 

Some time ago i suggested that Appgyver team should use complex projects to test the new releases and also for a better QA. I'm sure that most of the problems can't be detected using a very simple webview and a single page app. Please consider to use a more complex test app to enhance your tests... :)

**Comment by mmerkes** [comment]
@harsaharsa I ran our app with the latest edge build, and it seems to be working great! Thanks for getting this resolved.

**Comment by sebmartel** [comment]
@harsaharsa This is fantastic. I agree with @fidoboy and AppGyver could use our project for this purpose.

**Comment by fidoboy** [comment]
No issues in my project :) It seems that it works now :+1: great job guys!!

**Comment by bFlood** [comment]
did this affect the Web Platform build as well, or was it only a Crosswalk issue? thanks

**Comment by fidoboy** [comment]
@bFlood: you can try it and report back your results :) I'm using crosswalk and it seems to works fine.

As a side effect I've noticed that this release also solves this problem https://github.com/AppGyver/steroids/issues/657 it's not detailed on the changelog, but it seems that there was something related to cordova and crosswalk that now it's solved and makes that inAppBrowser now works as expected on Android!! cool! :)

**Comment by harsaharsa** [comment]
@bFlood this appeared to be a Crosswalk-only issue; at least I could not reproduce the bug (`devicready` not firing at all) on the Platform build.

@fidoboy great to hear – the fix had to do with how asset (JS/CSS etc) loading is handled, so it's possible that the same issue was affecting the `inAppBrowser` plugin and the requests it makes.

**Comment by fidoboy** [comment]
Yes, I was also having issues with the orientation sensors listeners and now they works fine. It seems that this fix has closed many related issues that I don't even expected. :)

**Comment by fidoboy** [comment]
There is however another unrelated issue (or may be it's related in some way with the assets loading on Android) that it's still not solved. The ```window.innerHeight``` or ```window.innerWidth``` aren't available on dom ready event on Android (it works without problems on iOS); both properties return 0 and there is no way to know the available screen size. So, sorry for the off topic here, but as i've said, may be it's related...

**Comment by djjolicoeur** [comment]
I have seen this several times w/ edge 4.0.5-edge7 android build:

````
03-10 10:34:59.830: I/chromium(19300): [INFO:CONSOLE(1147)] "Channel not fired: onFileSystemPathsReady", source: http://localhost/cordova.js (1147)
```

This results in our app not loading at all.  and happens after the 'device not ready after 5 seconds' log message as in the original issue.  It is intermittent, not consistent but it is still happening.

**Comment by harsaharsa** [comment]
@djjolicoeur how many WebViews/HTML assets are you loading? What device/Android version are you using? Can you share an example Steroids project where this is happening?

**Comment by djjolicoeur** [comment]
@harsaharsa , I cannot share the actual app with you as it is proprietary and I think the steps to spin up an app that would simulate what we are doing to try and reproduce would be non-trivial.  I was however, after several attempts able to reproduce the issue this morning on an emulator targeting android API level 19 and targeting the Google APIs platform 4.4.2, so I could capture the logs surrounding the event in their entirety.  We saw this several times yesterday on my coworkers Samsung Galaxy Tab 4 as well.  I do not have his device specifics at the moment, unfortunately.  

```
I/ActivityManager( 1266): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.vitalreactor.orchestra2/com.appgyver.runtime.standalone.chromium.MainActivity} from pid 1620
E/gralloc_goldfish(  937): gralloc_alloc: Mismatched usage flags: 400 x 640, usage 333
W/GraphicBufferAllocator(  937): alloc(400, 640, 1, 00000333, ...) failed -22 (Invalid argument)
E/        (  937): GraphicBufferAlloc::createGraphicBuffer(w=400, h=640) failed (Invalid argument), handle=0x0
E/BufferQueue( 1266): [ScreenshotClient] dequeueBuffer: SurfaceComposer::createGraphicBuffer failed
W/WindowManager( 1266): Screenshot failure taking screenshot for (400x640) to layer 21005
D/dalvikvm( 8659): Not late-enabling CheckJNI (already on)
I/ActivityManager( 1266): Start proc com.vitalreactor.orchestra2 for activity com.vitalreactor.orchestra2/com.appgyver.runtime.standalone.chromium.MainActivity: pid=8659 uid=10058 gids={50058, 3003, 1028, 1015}
D/dalvikvm( 8659): GC_FOR_ALLOC freed 62K, 6% free 2865K/3040K, paused 3ms, total 5ms
I/dalvikvm-heap( 8659): Grow heap (frag case) to 5.247MB for 2457612-byte allocation
D/dalvikvm( 8659): GC_FOR_ALLOC freed <1K, 4% free 5265K/5444K, paused 11ms, total 11ms
D/dalvikvm( 8659): GC_CONCURRENT freed <1K, 4% free 5265K/5444K, paused 1ms+0ms, total 2ms
D/com.appgyver.app.SteroidsApplication( 8659): SteroidsApplication created
D/com.appgyver.android.application.ApplicationActivity( 8659): (startUpUrl) ->  No startUpUrl present !
D/dalvikvm( 8659): GC_CONCURRENT freed 281K, 7% free 5674K/6068K, paused 18ms+0ms, total 20ms
D/com.appgyver.ui.ViewStack( 8659): Could not find the native css file: android.css from steroids application file path: native-styles/android.css
W/ResourceType( 8659): No known package when getting name for resource number 0x3ade68b1
D/com.appgyver.ui.ViewStack( 8659): createLoadingScreen()
W/XWalkInternalResources( 8659): org.xwalk.core.R$style.AutofillPopupWindow is not found.
W/XWalkInternalResources( 8659): org.xwalk.core.R$layout.autofill_text is not found.
W/XWalkInternalResources( 8659): org.xwalk.core.R$id.autofill_label is not found.
W/XWalkInternalResources( 8659): org.xwalk.core.R$id.autofill_menu_text is not found.
W/XWalkInternalResources( 8659): org.xwalk.core.R$id.autofill_popup_window is not found.
W/XWalkInternalResources( 8659): org.xwalk.core.R$id.autofill_sublabel is not found.
W/XWalkInternalResources( 8659): org.xwalk.core.R$drawable.autofill_popup_background is not found.
W/XWalkInternalResources( 8659): org.xwalk.core.R$drawable.autofill_popup_background_down is not found.
W/XWalkInternalResources( 8659): org.xwalk.core.R$drawable.autofill_popup_background_up is not found.
W/XWalkInternalResources( 8659): org.xwalk.core.R$dimen.autofill_text_divider_height is not found.
W/XWalkInternalResources( 8659): org.xwalk.core.R$dimen.autofill_text_height is not found.
W/XWalkInternalResources( 8659): org.xwalk.core.R$color.autofill_dark_divider_color is not found.
W/XWalkInternalResources( 8659): org.xwalk.core.R$color.autofill_divider_color is not found.
D/dalvikvm( 8659): Trying to load lib /data/app-lib/com.vitalreactor.orchestra2-1/libxwalkcore.so 0xb0fb2900
D/dalvikvm( 8659): Added shared lib /data/app-lib/com.vitalreactor.orchestra2-1/libxwalkcore.so 0xb0fb2900
I/LibraryLoader( 8659): Time to load native libraries: 2 ms (timestamps 2174-2176)
I/LibraryLoader( 8659): Expected native library version number "",actual native library version number ""
I/LibraryLoader( 8659): Expected native library version number "",actual native library version number ""
I/chromium( 8659): [INFO:library_loader_hooks.cc(104)] Chromium logging enabled: level = 0, default verbosity = 0
I/BrowserStartupController( 8659): Initializing chromium process, singleProcess=true
I/ResourceExtractor( 8659): Extracting resource icudtl.dat
I/ResourceExtractor( 8659): Extracting resource xwalk.pak
W/AudioManagerAndroid( 8659): Requires BLUETOOTH permission
D/        ( 8659): HostConnection::get() New Host Connection established 0xb8fc0bf0, tid 8659
W/chromium( 8659): [WARNING:proxy_service.cc(909)] PAC support disabled because there is no system implementation
D/dalvikvm( 8659): GC_CONCURRENT freed 589K, 11% free 5910K/6612K, paused 14ms+1ms, total 17ms
D/        ( 8659): HostConnection::get() New Host Connection established 0xb900cc90, tid 8690
I/chromium( 8659): [INFO:xwalk_extension_renderer_controller.cc(47)] EXTENSION PROCESS DISABLED.
I/CordovaLog( 8659): Changing log level to 2
D/JsMessageQueue( 8659): Set native->JS mode to 2
D/com.appgyver.ui.ViewStack( 8659): Pre Rendering  WebView: AGWebView -> URL: http://localhost/loading.html uuid: d68e7436-c1f1-4926-91d0-fc44a9aeab67
E/chromium( 8659): [ERROR:unix_domain_listen_socket_posix.cc(51)] Could not bind unix domain socket to com.vitalreactor.orchestra2_devtools_remote (with abstract namespace): Address already in use
D/PluginManager( 8659): init()
I/CordovaLog( 8659): Changing log level to 2
D/JsMessageQueue( 8659): Set native->JS mode to 2
D/com.appgyver.ui.TitleBarContainerView( 8659): presenting webview: AGWebView -> URL: http://localhost/index.html uuid: 7354c02b-a252-4219-9a0d-606de6a8b161
D/com.appgyver.ui.TitleBarContainerView( 8659): presenting webview: AGWebView -> URL: http://localhost/index.html uuid: 7354c02b-a252-4219-9a0d-606de6a8b161
D/dalvikvm( 8659): GC_EXPLICIT freed 659K, 12% free 6016K/6832K, paused 1ms+0ms, total 4ms
W/ResourceType( 8659): No known package when getting name for resource number 0x3ade6915
D/com.appgyver.ui.ViewStack( 8659): Will auto hide the splash screen
D/com.appgyver.app.CordovaFacade( 8659): Resuming the App
V/PushPlugin( 8659): onResume() -> webView: AGWebView -> URL: http://localhost/index.html uuid: 7354c02b-a252-4219-9a0d-606de6a8b161
V/PushPlugin-NotificationService( 8659): setForeground() -> oldValue: false newValue: true
D/com.appgyver.app.CordovaFacade( 8659): Resuming the App
D/com.appgyver.app.CordovaFacade( 8659): Resuming the App
V/PushPlugin( 8659): onResume() -> webView: AGWebView -> URL: http://localhost/loading.html uuid: d68e7436-c1f1-4926-91d0-fc44a9aeab67
W/EGL_emulation( 8659): eglSurfaceAttrib not implemented
D/OpenGLRenderer( 8659): Enabling debug mode 0
I/ActivityManager( 1266): Displayed com.vitalreactor.orchestra2/com.appgyver.runtime.standalone.chromium.MainActivity: +794ms
E/EGL_emulation( 8659): eglQuerySurface 30be  EGL_BAD_ATTRIBUTE
E/chromium( 8659): [ERROR:buffer_manager.cc(313)] [.BrowserCompositor-0xb91c1bb0]GL ERROR :GL_INVALID_OPERATION : glBufferData: <- error from previous GL command
E/EGL_emulation( 8659): eglQuerySurface 30be  EGL_BAD_ATTRIBUTE
I/chromium( 8659): [INFO:CONSOLE(1)] "exception firing the resume event from native. Error: ReferenceError: cordova is not defined", source:  (1)
I/chromium( 8659): [INFO:CONSOLE(1)] "exception firing the resume event from native. Error: ReferenceError: cordova is not defined", source:  (1)
D/com.appgyver.app.AGWebViewClientBase( 8659): onPageStarted(http://localhost/loading.html)
E/chromium( 8659): [ERROR:buffer_manager.cc(313)] [.BrowserCompositor-0xb9230a20]GL ERROR :GL_INVALID_OPERATION : glBufferData: <- error from previous GL command
D/com.appgyver.app.AGWebViewClientBase( 8659): onPageStarted(http://localhost/index.html)
D/CordovaNetworkManager( 8659): Connection Type: 3g
D/CordovaNetworkManager( 8659): Connection Type: 3g
D/com.appgyver.app.AGWebViewClientBase( 8659): onPageFinished(http://localhost/loading.html)
D/com.appgyver.ui.ViewStack( 8659): Pre Rendering complete for WebView: AGWebView -> URL: http://localhost/loading.html uuid: d68e7436-c1f1-4926-91d0-fc44a9aeab67
W/PluginManager( 8659): THREAD WARNING: exec() call to App.show blocked the main thread for 43ms. Plugin should use CordovaInterface.getThreadPool().
D/CordovaNetworkManager( 8659): Connection Type: 3g
D/CordovaNetworkManager( 8659): Connection Type: 3g
W/PluginManager( 8659): THREAD WARNING: exec() call to NetworkStatus.getConnectionInfo blocked the main thread for 44ms. Plugin should use CordovaInterface.getThreadPool().
I/chromium( 8659): [INFO:CONSOLE(26)] "Deprecation warning: moment().subtract(period, number) is deprecated. Please use moment().subtract(number, period).", source: http://localhost/javascripts/js/orchestra.js (26)
I/chromium( 8659): [INFO:CONSOLE(1154)] "deviceready has not fired after 5 seconds.", source: http://localhost/cordova.js (1154)
I/chromium( 8659): [INFO:CONSOLE(1147)] "Channel not fired: onFileSystemPathsReady", source: http://localhost/cordova.js (1147)
D/com.appgyver.app.AGWebViewClientBase( 8659): onPageFinished(http://localhost/index.html)
D/dalvikvm( 1266): GC_CONCURRENT freed 1478K, 47% free 6955K/12960K, paused 1ms+1ms, total 46ms
D/dalvikvm( 1266): GC_CONCURRENT freed 975K, 47% free 6916K/12960K, paused 1ms+1ms, total 40ms
D/dalvikvm( 1266): GC_CONCURRENT freed 907K, 47% free 6916K/12960K, paused 0ms+0ms, total 30ms
D/dalvikvm( 1266): GC_CONCURRENT freed 905K, 47% free 6915K/12960K, paused 1ms+37ms, total 39ms
```

**Comment by harsaharsa** [comment]
Is this happening with Crosswalk only, or platform WebView too?

In any case, you've likely nailed the cause at https://github.com/AppGyver/PushNotifications/issues/19. To go deeper:

We are using the `OnlineEventsBridgeMode`, which sets `jsMessageQueue.setPause(true)` during every `exec()` call. `jsMessageQueue.reset()` is called both in the `AGWebView` constructor as well as during `AGWebViewClient.onPageStarted()`. Each WebView has its own `PluginManager`, `NativeToJsMessageQueue` and `Cordova` instances, so at least the `reset()` calls shouldn't "bleed over" from other WebViews.

The bug that was causing very consistent "Channel not fired: onFileSystemPathsReady" errors was that in Crosswalk apps, the `onPageStarted()` function would be called for ALL resource loads, including JS assets and CSS assets, which would consistently reset the message queue. This happened because the `XWalkView` class doesn't implement a `onPageStarted()` method, just `onLoadStarted()`, which we piped to call `onPageStarted()`. This was clearly a mistake. :)

The initial bug is now fixed in the latest edge version (now we call `onPageStarted()` only if the loaded resource matches the current URL), but it seems that the solution is not 100% – looking more closely at your log, we can see that there's a `onResume()` event being fired to the WebView after which the `onPageStarted` is called, which would suggest that the `onPageStarted` event fires incorrectly the middle of Cordova initialisation, destroying the queue.

The actual `onPageStarted()` method in Crosswalk seems to be implemented by `XWalkViewClient` – we use our own `AGWebViewClient` class as the base, so I'll have to see if we can hook to that, or else find a way to trigger the `onPageStarted()` code at a place that won't interfere with the Cordova message queue. (Or not trigger it at all – since everything is created fresh when a new WebView is constructed, we might not even need to reset it separately; from the Cordova code I understand it's used for when you change `window.location` to a new page, which you shouldn't really do with Steroids anyway.)

**Comment by djjolicoeur** [comment]
Thanks @harsaharsa , I have not seen the "Channel Not Fired" error yet on edge9!  I do however still seem to have some issues with https://github.com/AppGyver/PushNotifications/issues/19 and have updated with some logs.  Again, more than willing to help out however I can.

**Comment by kaipk** [comment]
i actually tested my app with the  4.0.5-edge11 and i still get the deviceready not fired error or steroids.js not defined message. I am preloading 14 webviews. Huawei Honor, Android 4.4.2

**Comment by harsaharsa** [comment]
`steroids.js` not defined sounds like a separate issue, but the `deviceready` not fired event is likely caused by each WebView clearing the Cordova plugin queue when it loads. We have one possible solution to try for this, I'll keep you posted!

**Comment by djjolicoeur** [comment]
I am also seeing this on edge11

```
deviceready has not fired after 5 seconds. cordova.js:1154
Channel not fired: onFileSystemPathsReady 
```

followed by a white screen that requires me to force stop the app and restart.  This appears to have gotten worse since my last update on here.  I am now seeing ~%50 failure rate of the app to launch correctly.

**Comment by roberto-robles** [comment]
+1 Same thing here...

**Comment by kaipk** [comment]
@roberto-robles @djjolicoeur: do you also have the steroids.js not defined error message?

**Comment by djjolicoeur** [comment]
@kaipk I have not seen the steroids.js not defined message.

**Comment by harsaharsa** [comment]
My research into the `deviceready` bug today shows that it's not caused by the plugin queue being reset anymore (after the fix to not call `onPageStarted` on every asset load) – rather, each plugin executes normally and the WebView is informed that new results are available, but somehow the `cordova.js` running in the WebView doesn't evaluate the callback results. Curiously, initiating an UI change (i.e. pushing a layer) will cause the `deviceready` to fire – even after tens of seconds of not firing. This is likely the cause for the registration failure bug, too (https://github.com/AppGyver/PushNotifications/issues/19). Hunt continues.

**Comment by harsaharsa** [comment]
Android version `4.0.4-edge12` is now out that should fix the `deviceready` issue harder. I at least could not replicate it anymore after these fixes. Please confirm if it still occurs – hopefully promoting this to `4.0.5` stable soon. I've also added a bunch more logging so we can track down the issue better if it's still there.

**Comment by djjolicoeur** [comment]
Still seeing the channel not fired on edge12 build with a samsung galaxy s3

```
I/chromium( 8410): [INFO:CONSOLE(1154)] "deviceready has not fired after 5 seconds.", source: http://localhost/cordova.js (1154)
I/chromium( 8410): [INFO:CONSOLE(1147)] "Channel not fired: onFileSystemPathsReady", source: http://localhost/cordova.js (1147)
```

**Comment by kaipk** [comment]
i actually testet the edge12 now:

if i preload all my views the app crashes. if i preload only 2 views i got the <code>"Uncaught ReferenceError: steroids is not defined"</code> error message. The <code>"deviceready has not fired after 5 seconds"</code> error no longer appears. On IOS the app works without this message.

**Comment by harsaharsa** [comment]
@kaipk what do you mean by the problem being with the postMessage?

**Comment by kaipk** [comment]
@harsaharsa i send on each view an postMessage to check if the view is loaded. if all loaded i show the index view. i know now, if i using the steroids.js version 3.5.3 the apps works better but now i get the deviceready has not fired error message again. with the 3.5.10 the app crashes immediately after startup. if it helps you i can share my project with you.

**Comment by harsaharsa** [comment]
Instead of postMessage, could you maybe poll `steroids.getApplicationState()` and see if the preloads array is populated with all the correct views? This sounds like a sidetrack from the original thing (which is that `deviceready` won't fire even if nothing is done in any WebViews), so I'd suggest creating a new ticket for that with the repro.

**Comment by kaipk** [comment]
console.log(steroids.getApplicationState()); gives me undefined. any idea? but the deviceready not fired error is also present

**Comment by harsaharsa** [comment]
You have to call it like
```
steroids.getApplicationState(
  {},
  {
    onSuccess: function(applicationState) {
      console.log(applicationState)
    }
  }
);
```

Does the `deviceready` error happen on all projects, or just this one? Is it 100% consistent?

**Comment by kaipk** [comment]
Hi, now i am using the getApplicationState to check the loading state. if i preload more than 13 views the app crashes immediately after startup. the deviceready error happens not every time, but really often.

**Comment by harsaharsa** [comment]
The preloading 13+ views crash is a Crosswalk instability thing, see https://github.com/AppGyver/steroids/issues/698

Still not sure what's causing the `deviceready` error (haven't been able to reproduce it after the latest edge version), we're continuing to investigate.

**Comment by kaipk** [comment]
just for info: on every startup of the kitchensink app i got the deviceready error.

![bildschirmfoto 2015-04-08 um 09 14 49](https://cloud.githubusercontent.com/assets/851023/7040462/e0dc92f8-ddcf-11e4-8df1-a871452d8c30.png)

**Comment by pre** [comment]
@kaipk that might be caused by the SyntaxError (the first line). Javascript execution in the WebView sometimes goes nuts when some script errors like that.

The error above is caused by "file not found" .. the current Android webview interceptor does not properly return a status code, instead, when a Javscript file is not found, it's HTTP 200 but the contents of the javascript are `<html> .. Not Found .. </html>`. There the opening HTML tag for the "File not found" causes the SyntaxError (you can see it by clicking the "cloud-resrouces.js:1" on the upper right corner.

Recreate your `config/cloud-resources.raml` in case you can't get the cloud-resources.js generated correctly.

**Comment by kaipk** [comment]
@pre i used this qr-code. http://www.appgyver.com/supersonic/kitchensink
is this not not up to date?

**Comment by kaiserlos** [comment]
I am having the same, using a custom build scanner with platform view 4.0.5-e12 and a MPA on Android 4.4 .

On my index view it gets called but on my 3 to 4 preloaded views I am getting a 

deviceready has not fired after 5 seconds.
Channel not fired: onFileSystemPathsReady

if I show any of this preloaded views it gets executed immediately.

Somehow, it works sometimes after reloading the app. But this is only once in a while (without code change).

**Comment by harsaharsa** [comment]
I noticed the same thing – even in cases where deviceready was not firing for initially visible WebViews, it fired immediately when I e.g. navigated to a new view. It seems to be something to do with the JS online/offline event bridge (which is used by Android to notify the JS that new events are available) – everything seems to happen 100% without a hitch on the native side, but the messages just disappear when the JS should grab hold of them to handle the callbacks.

**Comment by laicosly** [comment]
+1

**Comment by aliyazdani** [comment]
+1

**Comment by kaiserlos** [comment]
I am back using 4.0.5 e1 with plattform view - seems to work but sometimes I have other problems with steroids getPicture() function. If I fire it sometimes nothing happens until I hit it again but then it crashs after making and selecting a picture.

**Comment by kaiserlos** [comment]
Tested it now with 4.0.5 custom scanner built and it still has this "random" problems. Once in a while all device ready calls are fired, even on preloaded webviews. Most of the time it is only fired on my index view. If I push a preloaded view it gets fired immediately. But still then most of the time some functions like a confirm prompt or the camera getpicture function are not working on the first call. If they work, they often don't call their callbacks, but if I hit the function (or another native function) the callback get suddenly fired. Feels similar to this: https://issues.apache.org/jira/browse/CB-6761 I also could observe this issue happening https://github.com/AppGyver/steroids/issues/783 

I tried a lot to track this issues with many log outputs on every call and logcat. But it also feels like sometimes it even works better if I am running in debugmode and watching on logcat.

This is very annoying and prevents us publishing important updates to our app. It is def not fixed in 4.0.5 and as the event bridge seems to connect many functions, this is an serious bug that should get fixed properly.

**Comment by harsaharsa** [comment]
This is still high on the backlog, we hope to have some more insight into this later this week. Sorry for the extreme delay in figuring this out.

**Comment by vsviridov** [comment]
Potentially related, we observed that if this error message shows up in the logs, there's a high probability of native communication results to be offset.

For instance, if we create a WebView and attach all event handlers, opening URL for the first time does not cause any callbacks to fire. Running the same code for the second time causes callbacks to fire, but the responses are from the first call. Calling it again yields responses from the second call, and so on.

Possibly has to do with how cordova callback id is generated in the cordova.callbacks object.

Same behaviour was observed with contact list API, when first call yields no callbacks, and second one get's the first call results, and so on.

**Comment by harsaharsa** [comment]
Ok, finally! We ended up rewriting the Cordova native bridge to directly evaluate the plugin results to the WebViews instead of using the online/offline event bridge, and it seems like it's working solidly. Huge apologies for the unreasonably long wait.

It's available from the Build Service as 4.0.6-edge2, please give it a try and tell us if things are fixed!

**Comment by vsviridov** [comment]
@harsaharsa With the 4.0.6-edge2 build, InAppBrowser plugin only fires 'exit' event. And also we've still seen the "device not ready..." message.

**Comment by kaipk** [comment]
@harsaharsa same problem width the 4.0.6-edge2 version. i got also the these error messages
```
deviceready has not fired after 5 seconds. cordova.js:1154
Channel not fired: onCordovaConnectionReady cordova.js:1147
Channel not fired: onCordovaInfoReady cordova.js:1147
```

**Comment by harsaharsa** [comment]
Could you please elaborate on your error cases (OS version, device etc) and provide a barebones repro project that exhibits the problem as a GitHub repo? The changes in 4.0.6-edge2 fixed our own repro cases on both Platform and Crosswalk WebViews (which was deviceready not sometimes firing on root views until you did a layers.push or tried to use a Cordova API), so we really need something to move forward with this.

@vsviridov can you make a new ticket about the InAppBrowser plugin errors, preferably with repro project/code?

**Comment by lemondesigns** [comment]
I found the issue regarding the deviceready matter resolved with 4.0.6-edge2, however I now have an issue with the native tab bar is unresponsive, one has to tap the screen several times (6 ot 7) before it eventually registers the tap and takes you to the relevant webview.

Is this something you're aware of?

**Comment by lemondesigns** [comment]
On further investigating, I note that the tabs are clickable, if you click on the actual "text" but the associated icon is not clickable - or does not register the click event.

**Comment by kaiserlos** [comment]
You are great! It took a long time for updates on this but now it is working on my side! Tested with adhoc plattform build, but I think internally it's a crosswalk build (take a look at my other issue). Also other callback functions ( camera getPicture, notification.confirm ....) are now working fine. Thank you very much, we can now finish our next app updates.

**Comment by lemondesigns** [comment]
@harsaharsa any ideas regarding the cause of the tab bar issue? Only the 'text title' is clickable on Android in 4.0.6-edge2 builds... the icon svg is not clickable and furthermore also does not change colour when that tab is active (only the text and the line above the tab button). It's quite irritating :(

**Comment by kaipk** [comment]
@harsaharsa i am using a Huawei Honor with Android 4.4.2. (Crosswalk, ARM only).

**Comment by harsaharsa** [comment]
@lemondesigns https://github.com/AppGyver/steroids/issues/956 for the tab bar issue, likely related to recent fixes to `background-size` property for tab bar images.

@kaipk, if I add many preloads, then the deviceready takes 5+ seconds to fire, but I can't replicate it not firing. The error is printed to console, but it does fire (e.g. after 5400ms).

@vsviridov, what device are you running on? Does it happen with a plain new project, or only after you do something (e.g. add multiple preloaded views etc)? Does Cordova still load – if you e.g. call `navigator.notification.alert("Hey!")` in a view that's reporting "deviceready not fired", does the native alert dialog show up? What about a `deviceready` event listener that just logs something to console?

As for the InAppBrowser issues, we're looking at them!

**Comment by JugnuPathak** [comment]
@harsaharsa Now cordova plugin callbacks are working fine in android 4.0.6-edge2 builds. 

Regarding to deviceready your right its firing perfectly but if we have numbers of preloads then its taking more than 5 sec. Its confusing people due to that console log.

**Comment by harsaharsa** [comment]
Yeah, we should probably bump it to like 10 seconds. I'll make a note.

**Comment by vsviridov** [comment]
It seems like supersonic data channels are very broken after this change. Inter-view communication happens when stars align. No consistency at all.

**Comment by harsaharsa** [comment]
@vsviridov what sort of structure do you have in your app? Does it break down with just two views, or only with more complex structures?

Could you create a new issue with repro steps for this so we can fix whatever might be wrong? Data channels use our own `postMessage` implementation and have nothing to do with Cordova, so it's strange that they'd be broken after this update.

**Comment by roberto-robles** [comment]
@harsaharsa for me 4.0.6-edge3 works like charm, we were having issues with edge2 and devicenotready and PushNotifications plugins issues. And now it seems to be working, our app only uses a single view, but loads a big amount of javascript so it's fairly complex. Good job with that!

**Comment by aliyazdani** [comment]
we tried 4.0.6 and the issue is still there.  I'm guessing once the project gets to a certain size this issue occurs.  When the project was small I think it happened less often but now it shows up every time.

**Comment by ubaidmohammed** [comment]
I am also having the same problem I Am on Cordova 3.5.0 have created the project from command line, for me the deviceOnReady is fired on the first page, page which is being called from the Activity JAVA class. But after from my first page I go to another HTML page the device onready is not being fired on that page. Any one has any clue about this?

**Comment by nikn** [comment]
try 
function onLoad() {
		document.addEventListener("deviceready", function () {//todo code}, false);
}

**Comment by mariobb** [comment]
Still same problem here...
I figured out that happend, to me, only the first time i open up the app. With no data nor cache.
On Android

**Comment by helloworld3q3q** [comment]
hello everyone ,how did you solve the problem? Can you please share?and I'm very worried. I use cordova version 6.2.0 and ionic  version 1.7.16 , and about android  android-minSdkVersion" value="21" .So  I get this problem, who can help me? thanks a lot

**Comment by aliyazdani** [comment]
We tried a lot of things but nothing worked.

**Comment by helloworld3q3q** [comment]
@aliyazdani  me too

**Comment by helloworld3q3q** [comment]
hello  everyone I solved the problem. the way  you should not  put any js files and html files to your server. when you use cordova emulate or cordova run  get  android apk.    *.js  and *.html in  your apk. Automatic operation inside the js file.  You only need to write a request for ip address .... example:  $http.get('http://110.110.2.110:8888/*.json).
简单说就是 打包时，你的js和html css 之类的已经在你的apk 里面了，服务器不上不需要放上js文件 和html css。 你只需要在ajax请求的时候请求时写对地址就好了。 额， 看来学好中文蛮重要的。

**Comment by helloworld3q3q** [comment]
I solved the problem, the way you do not need to js and html css placed on your server.  But you run cordova emulate or cordova run. the files have been placed inside the apk. So, you only need to write when ajax request to the server   address like  for example:  $ http.get ( 'http://192.110.2.110:800/test', function (data) {});  android apk will automatically load and run in the js.
简单的说就是不需要吧html js css 等放到你的服务器上，打包成功时，这些都在你的apk里面了，于是只需要写对 ajax请求地址就好。安卓程序会自动加载 apk里面的js 等文件的，然后进行http请求，这样做的好处，把文件放在本地，使程序启动更快，减少等待时间。额， 所以学好中文很重要。

**Comment by bayanAbuawad** [comment]
When you develop a hybrid application with PhoneGap, Cordova, Ionic, CordovaChromeApps and any other framework that is built on top on PhoneGap, I can almost assure you that you’ll get this error. And it’s super annoying.

I think the issue is when you delete your platforms folder, or the files inside, and do a platform add ... afterwards. This is at least how I reproduced it.

You’ll need to:

1) add the platforms with cordova platform add ios android

2) remove them with cordova platform remove ios android

3) add them again.

I know this is super weird, but I think it’s related to a faulty android.json and ios.json inside the platforms folder.

Following the steps above fixed the issue for me :).
